application/src/main/java/org/togetherjava/tjbot/commands/filesharing/FileSharingMessageListener.java

```diff
@@ -224,4 +248,74 @@ private boolean isHelpThread(MessageReceivedEvent event) {
         return isStagingChannelName.test(rootChannelName)
                 || isOverviewChannelName.test(rootChannelName);
     }
+
+    @Override
+    public String getName() {
+        return "filesharing";
+    }
+
+    @Override
+    public void acceptComponentIdGenerator(ComponentIdGenerator generator) {
+        componentIdInteractor.acceptComponentIdGenerator(generator);
+    }
+
+    @Override
+    public UserInteractionType getInteractionType() {
+        return UserInteractionType.OTHER;
+    }
+
+    @Override
+    public void onButtonClick(ButtonInteractionEvent event, List<String> args) {
+        Member user = event.getMember();
+        String authorId = args.get(0);
+        boolean hasSoftModPermissions =
+                user.getRoles().stream().map(Role::getName).noneMatch(softModPattern);
+
+        if (!authorId.equals(user.getId()) && hasSoftModPermissions) {
+            event.reply("You do not have permission for this action.").setEphemeral(true).queue();
+            return;
+        }
+
+        String gistId = args.get(1);
+        HttpRequest request = HttpRequest.newBuilder()
+            .uri(URI.create(SHARE_API + "/" + gistId))
+            .header("Accept", "application/json")
+            .header("Authorization", "token " + gistApiKey)
+            .DELETE()
+            .build();
+
+        HttpResponse<String> apiResponse;
+        try {
+            apiResponse = CLIENT.send(request, HttpResponse.BodyHandlers.ofString());
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new IllegalStateException(
+                    "Attempting to delete a gist, but the request got interrupted.", e);
+        }
+
+
+        int status = apiResponse.statusCode();
+        if (status == 404) {
+            throw new IllegalStateException("Gist API unexpected response while deleting gist: %s."
+                .formatted(apiResponse.body()));
```

Comment:
im not sure i like the UX in this case. right now it attempts to delete and then crashes with an exception. the user doesnt get any message and will just spam the button again and again. upon which it will constantly attempt again and crash more and more.

maybe it would be better to convert this into a `log.warn(...)`, tell the user that something went wrong and they should ask a moderator to delete it instead and disable the button as well.

Reply:
or u just disable the buttons right away, pretend that everything worked, and log the warning afterwards